File: E-Commerce-Front/e-commerce-front/src/app/module/OrderDetails.ts
import {Product} from './Product';
import {Order} from './Order';

export interface OrderDetails {
  id: number;
  products: Product[];
  order: Order;
  price: number;
  quantity: number;
  discount: number;
  total: number;
  size: number;
  color: string;
  isFulfilled: number;
  shipDate: Date;
  billDate: Date;
}

File: E-Commerce-Back/src/main/java/com/khalil/saidane/ecommerce/entities/OrderDetails.java
package com.khalil.saidane.ecommerce.entities;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Data
public class OrderDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToMany(mappedBy = "orderDetails")
    private List<Product> products;
    @OneToOne
    private Order_ order_;
    private float price;
    private int quantity;
    private float discount;
    private float total;
    private float size;
    private String color;
    private int isFulfilled;
    private Date shipDate;
    private Date billDate;
}
